app/serializers/credentials_serializer.rb

```diff
@@ -3,40 +3,40 @@ def initialize(user: , application:)
     @user, @application = user, application
   end
 
-  def serialize
-    {
-      user: serialized_user,
-      profile: serialized_profile,
-      roles: serialized_roles,
-    }
-  end
-
-  def to_json(opts = {})
-    serialize.to_json opts
+  def as_json(opts = {})
+    serialize
   end
 
   private
 
   attr_reader :user, :application
 
+  def serialize
+    {
+      "user" => serialized_user,
+      "profile" => serialized_profile,
+      "roles" => serialized_roles,
+    }
+  end
+
   def serialized_user
     {
-      email: user.email,
+      "email" => user.email,
     }
   end
 
   def serialized_profile
     {
-      email: profile.email,
-      name:  profile.name,
-      telephone: profile.tel,
-      mobile: profile.mobile,
-      address: {
-        full_address: profile.address,
-        postcode: profile.postcode,
+      "email" => profile.email,
```

Comment:
This syntax is to ensure that the keys are strings. @pixeltrix has mentioned before that yes I could just use `stringify_keys` at the end, and if that's the preference then I can do that. The hash rocket is necessary because doing `"email": value"` turns the key back into a symbol again. @pixeltrix has also said that the `as_json` method promises to return a hash with string keys, hence why it was done this way.